### PR TITLE
#9085 isNull Filter 

### LINKED ISSUE #9085: Query Panel isNull operator is buggy
## Description

The isNull operator in the query panel can not be applied alone

[isNull1.webm](https://user-images.githubusercontent.com/1279510/231113004-5f3fbff6-089c-4764-a8bc-17a0a6926c09.webm)

Event when used in couple, it is basically ignored on the map. 

[screencast-github.com-2023.04.11-11_17_06.webm](https://user-images.githubusercontent.com/1279510/231114672-e9dd3944-842c-4716-a075-909d64bb93d6.webm)

## How to reproduce

Bug 1: The isNull filter can not be applied
-  Add a vector layer to the map
- Open the query panel
- add a isNull filter condition on an attribute

*Expected Result*
- The filter can be applied

*Current Result*
- The filter can not be applied

Bug 2:  The isNull filter is ignored on the map.
- Add a vector layer to the map
- Open the query panel
- Add a isNull filter condition
- Add another attribute filter condition that should produce one or more result 
- Set the filter logic condition to "All" so the filter is something like ATTR isNull AND ATTR = "Alabama" so the filter should produce no results
- Apply the filter 
*Expected Result*
The map should show no result

*Current Result*
The map shows some results, as the "isNull" filter is not applied. 

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

### PR DESCRIPTION
**Pull request by offtherailz**
## Description

This PR fixes #9085 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#9085 

**What is the new behavior?**
Filter with only isNull can be applied. 
 isNull condition is not ignored anymore.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: build/tests.webpack.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
+var context = require.context('../web/client/utils', true, /(FilterUtils-test\.jsx?)|(-test-chrome\.jsx?)$/);
```
Comment (offtherailz) [inline build/tests.webpack.js:1]:
```suggestion
var context = require.context('../web, true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```

**Comment by ale-cristofori** [comment]
target branch for backport of this PR is
https://github.com/geosolutions-it/MapStore2/tree/c044-2022.02.xx

File: web/client/utils/__tests__/FilterUtils-test.js [hunk 2]
```diff
@@ -577,49 +578,65 @@ describe('FilterUtils', () => {
...
+            type: "string",
+            value: undefined
+        }, {
+            attribute: "attributeNull",
+            groupId: 1,
+            exception: null,
+            operator: "=",
+            rowId: "1",
+            type: "string",
+            value: null
```
Comment (MV88) [inline web/client/utils/__tests__/FilterUtils-test.js:632]: this seems duplicate of the first

File: build/tests.webpack.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
-var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
+var context = require.context('../web, true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```
Comment (MV88) [inline build/tests.webpack.js:1]:
```suggestion
var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```

**Review by MV88 — CHANGES_REQUESTED** [review]

File: web/client/utils/__tests__/FilterUtils-test.js [hunk 4]
```diff
@@ -577,49 +578,65 @@ describe('FilterUtils', () => {
...
+        }, {
```
Comment (offtherailz) [inline web/client/utils/__tests__/FilterUtils-test.js:633]:
```suggestion
        }, {
```

File: web/client/utils/__tests__/FilterUtils-test.js [hunk 5]
```diff
@@ -811,7 +828,24 @@ describe('FilterUtils', () => {
                 value: "isNull"
             }]
         };
-        let expected = '<wfs:GetFeature service="WFS" version="2.0" xmlns:wfs="http://www.opengis.net/wfs/2.0" xmlns:fes="http://www.opengis.net/fes/2.0" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/wfs/2.0 http://schemas.opengis.net/wfs/2.0/wfs.xsd http://www.opengis.net/gml/3.2 http://schemas.opengis.net/gml/3.2.1/gml.xsd"><wfs:Query typeNames="ft_name_test" srsName="EPSG:4326"><fes:Filter><fes:PropertyIsNull><fes:ValueReference>attributeNull</fes:ValueReference></fes:PropertyIsNull><fes:PropertyIsNull><fes:ValueReference>attributeValid</fes:ValueReference></fes:PropertyIsNull></fes:Filter></wfs:Query></wfs:GetFeature>';
+        let expected = '<wfs:GetFeature service="WFS" version="2.0" '
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/utils/__tests__/FilterUtils-test.js:823]: no, or spaces will not be the same.

**Review by MV88 — APPROVED** [review]

**Pull request merged**